glean-core/src/upload/mod.rs

```diff
@@ -195,8 +274,18 @@ impl PingUploadManager {
             .queue
             .write()
             .expect("Can't write to pending pings queue.");
-        match queue.pop_front() {
+        match queue.front() {
             Some(request) => {
+                // If unable to increment the counter,
+                // we have reached the rate limit of ping uploads for the time being.
+                if self.counter.increment().is_none() {
+                    log::info!(
+                        "Tried getting an upload task, but we are already at maximum processing capacity.
+                        Will come back later."
+                    );
+                    return PingUploadTask::Wait;
```

Comment:
Should we actually provide the amount of time to wait? Like `PingUploadTask::Wait(SECONDS_IN_A_MINUTE)`. I remember this being discussed back when `PingUploadTask` was first created. Could be useful in the future for other features we want to implement such as jitter delay.

Reply:
I believe we should, but maybe we don't need it just yet for this one.